ec2/instances.go

```diff
@@ -283,3 +283,28 @@ func (e *Ec2) RebootInstance(ctx context.Context, ids ...string) error {
 	}
 	return nil
 }
+
+func (e *Ec2) AttachVolume(ctx context.Context, input *ec2.AttachVolumeInput, attributeInput *ec2.ModifyInstanceAttributeInput) (string, error) {
+	if input == nil {
+		return "", apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+	log.Infof("Attaching volume of device %s", aws.StringValue(input.Device))
+
+	out, err := e.Service.AttachVolumeWithContext(ctx, input)
+	if err != nil {
+		return "", common.ErrCode("failed to attach volume", err)
+	}
+
+	log.Debugf("got output attach volume: %+v", out)
+
+	if out == nil {
+		return "", apierror.New(apierror.ErrInternalError, "Unexpected volume output", nil)
+	}
+
+	_, err = e.Service.ModifyInstanceAttributeWithContext(ctx, attributeInput)
```

Comment:
This should not be here but in the orchestrator. The `AttachVolume` function should only attach the volume